### DISCUSSION TITLE
Image optimization adapters

### DISCUSSION THREAD

**Discussion by BohdanK-W32** [payloadcms/payload#8499] [Feature Requests & Ideas]
When the image optimization adapters will be released? As I remember, it must've been happening before reaching the v3 stable?

The point is to use third-party services () and not to depend on sharp directly for such purposes.

Some of them, for example:
- [Cloudflare Images](https://www.cloudflare.com/developer-platform/cloudflare-images/)
- [Bytescale](https://www.bytescale.com)
- [Kraken.io](https://kraken.io/)

**Comment by sebastianmulders** [comment]
Curious to hear if there have been any updates on this!

**Reply by vonkoff** [reply]
Ditto

**Comment by followbl** [comment]
I hate `sharp`, and about 95% of the time I'm in the `payloadcms` is repo is to figure out why the latest or out of date `sharp` dependency is incompatible with `payloadcms`

it is a major PITA

we use imgix other places in our app and would love to completely swap this out

**Comment by zensitively** [comment]
imgix adapter would be very nice

**Comment by BohdanK-W32** [comment]
@denolfe what do you think about this one?
It looks like it's time to consider this one, after Cloudflare D1, R2, Workers already released.

**Reply by paulpopus** [reply]
This hasn't been prioritised yet but it would be something we can tackle after the current D1 and R2 adapters go stable from beta
